src/include/simeng/Core.hh

```diff
@@ -13,6 +14,40 @@ class ArchitecturalRegisterFileSet;
 
 enum CoreStatus : uint8_t { halted, idle, executing, switching };
 
+/** The result from a handled exception. */
+struct ExceptionResult {
+  /** Whether execution should halt. */
+  bool fatal;
+  /** The address to resume execution from. */
+  uint64_t instructionAddress;
+  /** Any changes to apply to the process state. */
+  simeng::OS::ProcessStateChange stateChange;
+};
+
+/** An abstract multi-cycle exception handler interface. Should be ticked each
+ * cycle until complete. */
+class ExceptionHandler {
```

Comment:
Would it be beneficial to add `This method...` or `Method to` to all function comments just to be stylistically consistent throughout the code base?